src/test/java/com/google/devtools/build/lib/remote/SimpleBlobStoreFactoryTest.java

```diff
@@ -0,0 +1,190 @@
+package com.google.devtools.build.lib.remote;
+
+import com.google.devtools.build.lib.remote.blobstore.CombinedDiskHttpBlobStore;
+import com.google.devtools.build.lib.remote.blobstore.OnDiskBlobStore;
+import com.google.devtools.build.lib.remote.blobstore.SimpleBlobStore;
+import com.google.devtools.build.lib.remote.blobstore.http.HttpBlobStore;
+import com.google.devtools.build.lib.vfs.Path;
+import com.google.devtools.build.lib.vfs.PathFragment;
+import com.google.devtools.common.options.Options;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.IOException;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+@RunWith(JUnit4.class)
+public class SimpleBlobStoreFactoryTest {
+
+  @Rule
+  public ExpectedException expectedEx = ExpectedException.none();
+
+  @Mock
+  private Path workingDirectory;
+
+  @Mock
+  private Path relativeDirectory;
```

Comment:
Instead of mocking these consider using real path objects. It's quite straightforward. Here is an example of how to do it: https://source.bazel.build/bazel/+/master:src/test/java/com/google/devtools/build/lib/remote/GrpcRemoteExecutionClientTest.java;l=229

Reply:
Hm, it is, but I would argue that mock is perfectly fine on its place in this test case - we can emulate behavior we need and also verify the calls made from the class under tests to our mock.

Reply:
Prefer to use a real object or a fake if you can and only if that doesn't work use a mock. Besides that, using mocks for these tests is in fact problematic because it ties the test to the implementation of `SimpleBlobStoreFactory.create` while it should just test the behavior of it. Consider this code from `createCombinedCache()`:

```
when(workingDirectory.getRelative(remoteOptions.diskCache)).thenReturn(relativeDirectory);
when(relativeDirectory.exists()).thenReturn(true);
```

There are multiple ways how `SimpleBlobStoreFactory.create` could be implemented. For example, instead of calling `getRelative` it could first check if the path fragment is absolute (which in fact it is in this test) and if so directly create the path. This test would then fail.

Instead of calling `Path.exists()` it could call `Path.stat()`. It could call `Path.isWriteable()/isReadable()` to ensure it can properly use this directory and also call `Path.chmod` on it.

Reply:
Fair point. Will do